migration/migrator/migrations/master/202306205342134_add_course_and_semester_constraint.py

```diff
@@ -0,0 +1,8 @@
+def up(config, database):
+    database.execute("ALTER TABLE courses ADD CONSTRAINT no_surrounding_spaces_course CHECK (course ~ '^[^ ].*[^ ]$')")
+    database.execute("ALTER TABLE terms ADD CONSTRAINT no_surrounding_spaces_semester CHECK (term_id ~ '^[^ ].*[^ ]$')")
+
+def down(config, database):
+    database.execute("ALTER TABLE courses DROP CONSTRAINT no_surrounding_spaces_course")
+    database.execute("ALTER TABLE terms DROP CONSTRAINT no_surrounding_spaces_semester")
```

Comment:
`up()` and `down()` should each contain a single `database.execute()` call to make sure the entire migration gets rolled back as a single transaction if something goes wrong.